lib/jenkins_api_client/client.rb

```diff
@@ -320,6 +299,30 @@ def make_http_request(request, follow_redirect = @follow_redirects)
     end
     protected :make_http_request
 
+    # Downloads a url to a filenam
+    #
+    # @param [String] url The url to download
+    # @param [String] filename The filename of the file to store the download to
+    #
+    def download_url(url, filename)
+      uri = URI.parse(url)
+      @logger.info "Starting download for: #{uri.to_s}"
+      http_object = Net::HTTP.new(uri.host, uri.port)
+      http_object.use_ssl = true if uri.scheme == 'https'
+      http_object.start do |http|
+        request = Net::HTTP::Get.new uri.request_uri
+        http.read_timeout = 600
+        http.request request do |response|
+          open filename, 'w' do |io|
+            response.read_body do |chunk|
+              io.write chunk
+            end
+          end
+        end
+      end
+      @logger.info "Stored download as " + filename + "."
```

Comment:
Let's change t his to `@logger.info "Stored download as #{filename}."` for consistency.